architecture/adr-014-module-catalog.md

```diff
@@ -0,0 +1,28 @@
+<!-- Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. -->
+
+| Type          | ID            | Status        | Title                   |
+| ------------- | ------------- | ------------- | ----------------------- |
+| ADR           | adr-014       | Accepted      | Multiple ModuleCatalogs |
+
+
+# Architecture Decision Record: Multiple ModuleCatalogs
+
+## Context
+ComposeUI aims to support many different kind of modules. These modules, that can be started by the ModuleLoader should be stored in ModuleCatalogs and every module should have its unique module id. The user should have the ability to handle multiple ModuleCatalogs with the ModuleLoader.
+
+General problem statement:
+- When multiple ModuleCatalogs are used, some modules could have the same module id in different ModuleCatalogs which could cause issues while starting or stopping the module via the ModuleLoader.
+
+## Decision
+- If the same module id is being used for multiple modules (likely in different ModuleCatalogs), the first occurrence will be used. In case of multiple ModuleCatalog registrations, the order of the registration determines which module can be used.
```

Comment:
[ZKRobi](https://github.com/ZKRobi) [4 days ago](https://github.com/morganstanley/ComposeUI/pull/569#discussion_r1555944683)
The same id in the same catalog shouldn't ever exist, although it's fine to describe the behavior.
I'd specify what is the first occurance in case of multiple catalogs though -> e.g. in the order of provider registration sounds like a method that is easy to understand, and provides some agency to the developer. This might have implications on how to implement merging modulecatalogs in case of async fetching.